lib/ansiblelint/__init__.py

```diff
@@ -19,167 +19,9 @@
 # THE SOFTWARE.
 """Main ansible-lint package."""
 
-from collections import defaultdict
-import logging
 import os
 
-from ansiblelint.rules import AnsibleLintRule  # noqa F401: exposing public API
-import ansiblelint.utils
-import ansiblelint.skip_utils
-from ansiblelint.errors import MatchError
-from ansiblelint.rules.LoadingFailureRule import LoadingFailureRule
-from typing import List, Set
+# package should not do any imports, especially those that would have side
+# effect of importing ansible.
```

Comment:
We should postpone removing the public API imports.

Reply:
AFAIK that is not possible. Breaking the API is mandatory as part of cleaning the   `__init__py` file. It is not possible to keep it compatible and avoid running undesired imports chain.

We need to document the change well, allowing custom rule users. Also that is the kind of change that cannot be made in a minor release, so 4.3.0 is the right kind of release to do it.

I challenge you to find a way to avoid id. I am quite curious as dislike breaking API too. The reality is that AnsibleLintRule class loading cannot happen without loading ansible modules. If this happens in ``__init__``, we loose ability to run ``__main__`` module w/o already having ansible imported.

If we move the bootstrapping our of __main__ to init we loose the ability to perform boostrapping using config options or logger, both needed.